### PR TITLE
Feature/add logs

### PR DESCRIPTION
**Pull request by Nalien**
Add logging to the application + config

Updated the Tests to have one Init method

### CODE REVIEW

File: business/MetadataDatabase/Services/PacsMirrorService.cs [hunk 1]
```diff
@@ -27,11 +30,13 @@ public PacsMirrorService(ISeriesServices seriesService, IPacsService pacsService
         /// </summary>
         public void MirrorPacs()
         {
+            logger.LogInformation("Start miroring PACS");
```
Comment (cquennouelle) [inline business/MetadataDatabase/Services/PacsMirrorService.cs]: mirroring :)

File: business/MetadataDatabase/Services/PacsMirrorService.cs [hunk 2]
```diff
@@ -41,15 +46,20 @@ public void MirrorPacs()
             foreach (string uid in seriesUidsNotInPacs)
             {
                 IEnumerable<SeriesDto> query = dataBaseSeries.Where(series => series.SeriesInstanceUID == uid);
-                this.seriesService.Delete(query.FirstOrDefault().Id);
+                var serieIdToRemove = query.FirstOrDefault().Id;
+                logger.LogInformation($"Removed serie : {serieIdToRemove}");
```
Comment (cquennouelle) [inline business/MetadataDatabase/Services/PacsMirrorService.cs]: s/serie/series

File: business/MetadataDatabase/Services/PacsMirrorService.cs [hunk 3]
```diff
@@ -41,15 +46,20 @@ public void MirrorPacs()
...
+                this.seriesService.Delete(serieIdToRemove);
             }
             // Create into DB all series not in the DB but in PACS
             IEnumerable<string> seriesUidsNotInDb = pacsSeriesUids.Except(dataBaseSeriesUids);
             foreach (string uid in seriesUidsNotInDb)
             {
                 IEnumerable<SeriesDto> query = pacsSeries.Where(series => series.SeriesInstanceUID == uid);
+                logger.LogInformation($"Added serie : {uid}");
```
Comment (cquennouelle) [inline business/MetadataDatabase/Services/PacsMirrorService.cs]: s/serie/series

**Review by cquennouelle — APPROVED** [review]

File: business/MetadataDatabase/nlog.config [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  <!-- enable asp.net core layout renderers -->
+  <extensions>
+    <add assembly="NLog.Web.AspNetCore"/>
+  </extensions>
+
+  <!-- the targets to write to -->
+  <targets>
+    <!-- write logs to file  -->
+    <target xsi:type="File" name="allfile" fileName="./logs/nlog-all-${shortdate}.log"
+            layout="${longdate}|${event-properties:item=EventId_Id}|${uppercase:${level}}|${logger}|${message} ${exception:format=tostring}" />
```
Comment (cquennouelle) [inline business/MetadataDatabase/nlog.config:17]: Ajouter des whitespaces, ce serait une bonne idée ou de la place perdue ?

**Review by cquennouelle — DISMISSED** [review]

**Review by bat3 — DISMISSED** [review]
With Cyril comments ;-)

**Review by ptitdam2001 — DISMISSED** [review]
I approve but I have one question: Do logs be available in Docker logs?

File: business/MetadataDatabase/Services/PacsService.cs [hunk 5]
```diff
@@ -14,18 +14,18 @@ namespace MetadataDatabase.Services
...
         public PacsService(IOptions<PacsConfiguration> settings)
         {
             this.settings = settings.Value;
+            this.client = new HttpClient();
         }
 
         public IEnumerable<SeriesDto> GetSeriesList()
         {
             Task<IEnumerable<QidoSeries>> task = this.GetSeriesListAsync();
-            task.Wait();
```
Comment (Nalien) [inline business/MetadataDatabase/Services/PacsService.cs:28]: task.await(à is included in task.Result

**Review by bat3 — APPROVED** [review]

**Review by cquennouelle — APPROVED** [review]

**Pull request merged**